infrastructure/terraform/modules/client_scopes/main.tf

```
@@ -38,34 +36,17 @@ resource "null_resource" "apply_custom_oid4vc_client_scopes" {
         -d "password=$KC_ADMIN_PASS" \
         -d "grant_type=password" | jq -r .access_token)
 
-      echo "Importing OID4VC client scopes via curl..."
-
-      # Import IdentityCredential scope
-      cat <<EOF | curl -k -s -X POST "$KC_URL/admin/realms/${var.realm_name}/client-scopes" \
-        -H "Authorization: Bearer $TOKEN" \
-        -H "Content-Type: application/json" \
-        --data-binary @-
-      ${local.identity_credential_json}
-      EOF
-
-      # Import SteuerberaterCredential scope
-      cat <<EOF | curl -k -s -X POST "$KC_URL/admin/realms/${var.realm_name}/client-scopes" \
-        -H "Authorization: Bearer $TOKEN" \
-        -H "Content-Type: application/json" \
-        --data-binary @-
-      ${local.steuerberater_credential_json}
-      EOF
+      echo "Importing OID4VC client scope from ${each.value} via curl..."
 
-      # Import KMACredential scope
-      cat <<EOF | curl -k -s -X POST "$KC_URL/admin/realms/${var.realm_name}/client-scopes" \
+      # Import the client scope
+      curl -k -s -X POST "$KC_URL/admin/realms/${var.realm_name}/client-scopes" \
```

Could you consider adding validation to the responses from curl commands?
Possible here others necessary?
For example: 
- Retrieving a token but not validating it. If authentication fails, the script continues and fails later. We can prevent silent failures and unclear errors.
- Missing error check for client scope creation 

Sample
```tf
if [ -z "$TOKEN" ] || [ "$TOKEN" = "null" ]; then
  echo "Failed to obtain admin token" >&2
  exit 1
fi
```